openedx/core/djangoapps/programs/tasks/v1/tasks.py

```diff
@@ -194,8 +194,14 @@ def award_program_certificates(self, username):
                 )
             except Exception:  # pylint: disable=broad-except
                 # keep trying to award other certs, but retry the whole task to fix any missing entries
-                LOGGER.exception('Failed to award certificate for program %s to user %s', program_uuid, username)
-                retry = True
+                warning_message = 'Failed to award certificate for program {uuid} to user {username}.'.format(
+                    uuid=program_uuid, username=username)
+
+                if countdown < MAX_RETRIES:
```

Comment:
@tasawernawaz That's not how this works.  The retry method handles this logic for you.

countdown is a time delay and MAX_RETRIES is a number of attempts.  These cannot be compared.
http://docs.celeryproject.org/en/latest/reference/celery.app.task.html#celery.app.task.Task.retry

Reply:
@bderusha if retry logic already handling this. I think I only need to log message as warning, once retries are exceeded, error will be logged from retry logic ?